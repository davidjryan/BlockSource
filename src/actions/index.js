import firebase from 'firebase'
import { auth } from '../services/firebase'
import { fetchStoredProjects, fetchProjectContributors } from '../services/api'

/*------------------ Fetch Endpoints --------------------------- */

export const projectsEndpoint = () => async (dispatch) => {
    const projects = await fetchStoredProjects();
    dispatch(setProjects(projects));
};

export const setProjects = (projects) => {
  return {
    type: 'SET_PROJECTS',
    projects
  }
};

export const contributorsEndpoint = (id) => async (dispatch) => {
  const contributors = await fetchProjectContributors(id);
  dispatch(setProjectContributors(contributors))
}

export const setProjectContributors = (contributors) => {
  return {
    type: 'SET_CONTRIBUTORS',
    contributors
  }
};

/*------------------ Login actions --------------------------- */

export const createUser = (email, password) => async (dispatch) => {
  auth.createUserWithEmailAndPassword(email, password).then((user) => {
    dispatch(createUserSuccess(user))
  }).catch((error) => {
    dispatch(createUserError(error))
  })
};

export const createUserSuccess = () => {
  return {
    type: 'CREATE_USER_SUCCESS'
  }
};

export const createUserError = () => {
  return {
    type: 'CREATE_USER_ERROR'
  }
};

/*----------------- Search Actions --------------------------- */

export const searchInputChange = (searchInput) => {
  return {
    type: 'SEARCH_CHANGE',
    searchInput
  }
};

export const clearSearchInput = () => {
  return {
    type: 'SEARCH_CLEAR',
    searchInput: ''
  }
};

