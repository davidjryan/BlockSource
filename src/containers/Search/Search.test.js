import React from 'react';
import { shallow } from 'enzyme';
import * as actions from '../../actions'
import { Search, mapStateToProps, mapDispatchToProps } from './Search';

describe('Search tests', () => {
  let mockSearchInput;
  let inputChange;
  let mockProjects;
  let search;

  beforeEach( () => {
    mockSearchInput = 'hoverboard';
    inputChange = jest.fn();
    mockProjects = [{title: 'hoverboard'}, {title: 'flying cars'}]

    search = shallow(<Search 
                        searchInput={mockSearchInput}
                        projects={mockProjects}
                        inputChange={inputChange}
                      />)
  })

  it('should match the snapshot', () => {
    expect(search).toMatchSnapshot()
  })

  describe('mapStateToProps tests', () => {
    it('should pull projects from the store', () => {
      const mockSearchInput = {value: 'hoverboard'}
      const mockStore = {
        projects: mockProjects,
        searchInput: mockSearchInput.value
      };

      actions.searchInputChange('hoverboard')
      const result = mapStateToProps(mockStore);
      console.log(mockStore)
      console.log(result)

      expect(result.projects).toEqual(mockStore.projects[0]);
    });
  });

  describe('mapDispatchToProps tests', () => {
    it('Should call dispatch when inputChange is called', () => {
      const mockDispatch = jest.fn();
      const result = mapDispatchToProps(mockDispatch);

      result.inputChange('hoverboard');
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})