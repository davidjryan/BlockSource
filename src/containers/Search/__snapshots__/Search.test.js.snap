// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    inputChange={[Function]}
    projects={
        Array [
            Object {
              "title": "hoverboard",
            },
            Object {
              "title": "flying cars",
            },
          ]
    }
    searchInput="hoverboard"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search-bar"
        onChange={[Function]}
        placeholder="Search"
        type="text"
/>,
      "className": "search-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search-bar",
        "onChange": [Function],
        "placeholder": "Search",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-bar"
          onChange={[Function]}
          placeholder="Search"
          type="text"
/>,
        "className": "search-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-bar",
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
